units/quotaon@.service.in

```diff
@@ -0,0 +1,23 @@
+#  SPDX-License-Identifier: LGPL-2.1-or-later
+#
+#  This file is part of systemd.
+#
+#  systemd is free software; you can redistribute it and/or modify it
+#  under the terms of the GNU Lesser General Public License as published by
+#  the Free Software Foundation; either version 2.1 of the License, or
+#  (at your option) any later version.
+
+[Unit]
+Description=Enable File System Quotas
+Documentation=man:quotaon(8)
+
+ConditionPathExists={{QUOTAON}}
+
+DefaultDependencies=no
+After=remote-fs.target
```

Comment:
Why `After=remote-fs.target` is necessary?

Reply:
This was to get the ordering of quotaon after the remote fs mount right.
I didn't find a way to correctly add the ordering directly from the instantiated mount unit to the instantiated quotaon unit. 
I could only add ordering for the template, but that would incorrectly order instantiated quotaon units to the instantiated mounts.
Therefore I opted to order it after the target, which makes sure that all mounts are there when quotaon is running.

Reply:
this is not ok. we should have precise dependencies. file systems are mounted at various times, not just at boot, and we should ave proper ordering then too.

moreover, we should have sensible ordering between systemd-makes@.service, systemd-fsck@.service, the actual mount, systemd-growfs@.service, systemd-pcrfs@.service, systemd-quotacheck@.service and systemd-quotaon@.service.

I'd like to understand what trouble you specifically had? is this about the fact that quota wants a device node rather than an fs path? if so, maybe add a wrapper that translates an fspath to a device path. It's trivial, we already have helpers for that, look for `get_block_device()`. 

I am pretty sure that any of our services that run for a file system before the mount should take a device node as parameterization, but those after the mount should only take a file system path

Reply:
Actually, the failing dependencies on generated units were my mistake.
It works setting dependencies from the generated mount unit to the generated quota units.
With the current commit, I get correct results in my tests.
The failing ci runs appear to be unrelated to my patch.
Ok, like that?